src/mqtt.cpp

```diff
@@ -61,14 +62,16 @@ MQTT::MQTT(AsyncWebServer* server, DNSServer *dns, const char* MqttServer, uint1
 
   } else {
     // use Wifi
-    improvSerial.ConnectToWifi();
+    if(improvSerial.ConnectToWifi()) {if (Config->GetDebugLevel() >=4) WiFi.printDiag(dbg);} // printDiag() nur ausführen, wenn ConnectToWifi() OK ist
+    else                             {dbg.printf("ConnectToWiFi() failed\n");}
   }
 
-  if (Config->GetDebugLevel() >=4) WiFi.printDiag(dbg);
-
   dbg.printf("Initializing MQTT (%s:%d)\n", Config->GetMqttServer().c_str(), Config->GetMqttPort());
   espClient = WiFiClient();
-  
+
+  #ifdef JSON_MQTT
```

Comment:
es darf kein grosser buffer aufgebaut werden, jedes topic muss alleine für sich unabhängig gesendet werden. Bei grossen Definitionen wird einfach das json abgeschnitten und ungültiger code wird gesendet. im schlimmeren fall bootet der esp neu

Reply:
doch, dafür habe ich in denn modbus-Konstruktor oben
```c++
this->mqttJson.reserve(JSON_MSG_BUFFER);
``` 
aufgenommen, um einen 4,1k-Buffer zu haben. Das beseitigt abgeschnittene JSONs und beseitigt Abstürze (habe ich selbst schmerzhaft erarbeitet... :). Wenn man über meinen Weg gnadenlos alles ausgibt, treten die Probleme natürlich irgendwann wieder auf, wenn 4k Nutzdaten erreicht sind.

Und mein 2019er FHEM auf einem rasPi2 fraß sich mit einzeln gesendeten Topics nach 2-3Tagen zuverlässig fest - ein Verhalten, das mit der großen JSON-Antwort nie auftrat. Aber vielleicht ist das System ja auch etwas aus der Zeit gefallen...